src/msftoolbox/reliefweb/reports.py

```diff
@@ -0,0 +1,253 @@
+        }
+
+        payload = {
+            "preset": self.preset,
+            "limit": self.limit,
+            "query": {
+                "value": query_value,
+                "fields": query_fields,
+                "operator": query_operator
+            },
+            "filter": all_filters,
+            "fields":{"include":["source.name", "date", "language.name"]}
+        }
+
+        # Send request to URL
+        response = requests.post(
+            base_url,
+            headers=headers,
+            json=payload
+            )
+
+
+        if response.status_code != 200:
+            raise requests.HTTPError(
+                f"We have an error with status code: {response.status_code} \
+                and text: {response.text}"
+            )
+
+        if response_format == "dictionary":
+            response_data = response.json()["data"]
+            records = [
+                {
+                    "id": record["id"],
+                    "title": record["fields"]["title"],
+                    "title_english": record["fields"]["title"],
+                    "source_name": " / ".join([source['name'] for source in record["fields"]["source"]]),
+                    "language": " / ".join([language['name'] for language in record["fields"]["language"]]),
+                    "date": record["fields"]["date"]["original"],
+                    "url": record["href"]
+                } for record in response_data
+            ]
+            return records
+        else:
+            return pd.DataFrame(response.json()["data"])
+
+
+    def get_report(
+        self,
+        report_url:str
+        ) -> dict:
+        """Get a report based on the report url.
+
+        Args:
+            report_url (str): The url for the report. Typically the href the list_reports response
+
+        Returns:
+            dict: A dictionary containing keys the full report information.
+        """
+        profile = self.profile
+        base_url = f"{report_url}?appname={self.app_name}&profile={profile}"
+
+        # Send request to URL
+        response = requests.get(
+            base_url
+            )
+
+        if response.status_code != 200:
+            raise requests.HTTPError(
+                f"We have an error with status code: {response.status_code} \
+                and text: {response.text}"
+            )
+
+        return response.json()["data"]
+
+    def parse_report_data(
+        self,
+        report_data:dict
+        ) -> dict:
+        """Returns a subset of fields. The ID of the report, the title, the status, \
+        the body, the original link and the url.
+
+        Args:
+            report_data (dict): The response from get_report
+
+        Returns:
+            dict: A dictionary containing keys id, title, status, \
+             origin_url, source_url and country
+        """
+
+        report_fields = report_data[0]["fields"]
+        fields_to_subset = {
```

Comment:
line 235 to 248. Do we want to inforce a subset schema or retain the API response? 

Enforcing the schema was good for the media monitor because we wanted to standardise across various sources, but maybe here it would be better to leave the full API response and let the user handle the schema.

Reply:
So I would say that possibly, we remove the parse_report_data method.

Reply:
agreed